### PR TITLE
[Kotlin Migration] ImportUtils

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
```
> Task :AnkiDroid:compilePlayReleaseKotlin
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt: (211, 51): Smart cast to 'String' is impossible, because 'mFileName' is a mutable property that could have been changed by this time
```

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt [hunk 1]
```diff
@@ -67,394 +58,364 @@ public class ImportUtils {
...
+                sendShowImportFileDialogMsg(tempOutDir)
+                ImportResult.fromSuccess()
             }
         }
 
-        @Nullable
-        protected ImportResult validateZipFile(Context ctx, String filePath) {
-            File file = new File(filePath);
-            ZipFile zf = null;
+        private fun validateZipFile(ctx: Context, filePath: String?): ImportResult? {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt:189]: `KotlinCleanup`: nonnull path

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt [hunk 2]
```diff
@@ -67,394 +58,364 @@ public class ImportUtils {
...
+                context.contentResolver.openInputStream(data!!)
+            } catch (e: FileNotFoundException) {
+                Timber.e(e, "Could not open input stream to intent data")
+                return false
             }
             // Check non-null
-            if (in == null) {
-                return false;
+            @Suppress("FoldInitializerAndIfToElvis")
+            if (`in` == null) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt:283]: `kotlinCleanup`: rename

**Comment by Akshay0701** [comment]
yes solving....

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt [hunk 3]
```diff
@@ -67,394 +58,364 @@ public class ImportUtils {
...
 
-        public static ImportResult fromErrorString(String message) {
-            return new ImportResult(message);
-        }
+        companion object {
+            /**
+             * Send a Message to AnkiDroidApp so that the DialogMessageHandler shows the Import apkg dialog.
+             * @param path path to apkg file which will be imported
+             */
+            private fun sendShowImportFileDialogMsg(path: String?) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt:306]: kotlinCleanup: nonnull path

File: AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt [hunk 4]
```diff
@@ -67,394 +58,364 @@ public class ImportUtils {
...
-        public static ImportResult fromInvalidZip(Context ctx, File file, Exception e) {
-            return fromErrorString(getInvalidZipException(ctx, file, e));
+            /**
+             * Check if the InputStream is to a valid non-empty zip file
+             * @param data uri from which to get input stream
+             * @return whether or not valid zip file
+             */
+            private fun hasValidZipFile(context: Context, data: Uri?): Boolean {
+                // Get an input stream to the data in ContentProvider
+                var `in`: InputStream? = null
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/ImportUtils.kt:349]: `in` again

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
Should save a second review cycle

**Review by david-allison-1 — APPROVED** [review]
LGTM

**Review by mikehardy — APPROVED** [review]

**Pull request merged**